### PR TITLE
Export annotations

### PR DESCRIPTION
**Pull request by jufabeck2202**
I wanted to use the trained model in python script wich required the frozen_inference_graph.pb. Because I didn't want to download the whole bucket every time, I added an export --graph option to only export the frozen_inference_graph.pb, label_map.pbtxt and the training-log.txt. After exporting I move the files from the training to the model folder so the user can identify the files.

I also added an option to export only the pictures and annotations using the export --annotations keyword.

### CODE REVIEW

**Comment by bourdakos1** [comment]
This is a good idea! However, I feel it would be better suited to move this code to `cacli download <model-id>`. The download script has the context of the model-id, so it will be much easier to get the right files.

I'm thinking we could change `cacli download <model-id>` to download all of the training output for the given run by default (checkpoints, frozen graph, logs, tflite model, core ml model, etc..) and you could run `cacli download <model-id> --coreml --graph --file=training-log.txt` to narrow the download to what you want?

**Comment by jufabeck2202** [comment]
Yes, you are right, that sounds like a better place for it. 
So by default when you execute `cacli download <model-id>` it downloads model_ios, model_web, model_andoid and model folder, which will contain the checkpoints, frozen graph, logs, label map, and other training data.
And if you want only the iOS model you can use `cacli download <model-id> --coreml` to download only it.

**Comment by bourdakos1** [comment]
At the moment cacli download runs this:
```
await downloadDir(cos, bucket, model_location, 'model_ios')
await downloadDir(cos, bucket, model_location, 'model_web')
await downloadDir(cos, bucket, model_location, 'model_android')
```

Instead the default should just be:
```
await downloadDir(cos, bucket, model_location)
```

And yes, if they specify a flag like `--coreml`, it will only download the iOS model

**Comment by jufabeck2202** [comment]
I now changed the download options to
```
cacli download <model_id>              Download all Models
cacli download <model_id> --tflite     Download tflite Model for Android
cacli download <model_id> --coreml     Download coreml Model for iOS
cacli download <model_id> --web        Download web Model
```
By default it will download the whole trainig folder and save it under `./<model_id>/` . That way you can still identify your model after download. So when executing `cacli download <model_id> --tflite --coreml` it will save under
```
./<model_id>/
	model_ios/
	model_android/
```

**Comment by bourdakos1** [comment]
This looks great! I look over it more closely later today and merge it if everything looks 100%

File: src/commands/dump.js [hunk 1]
```diff
@@ -40,6 +40,24 @@ const downloadBucket = async (cos, bucket, path) => {
   await Promise.all(promises)
 }
 
+const downloadAnnotations = async (cos, bucket, path) => {
+  const files = await fileList(cos, bucket)
+  const promises = files.filter(file=>{
+    //filter for jpegs and annotation.json
+    return (/\.(jpe?g)$|_annotation.json/gim.test(file))
```
Comment (bourdakos1) [inline src/commands/dump.js]: Just for consistency, this is the regex I use on the actual client:
```js
const IMAGE_OR_ANNOTATIONS_REGEX = /\.(jpg|jpeg|png)$|^_annotations.json$/i
files.filter(file => IMAGE_OR_ANNOTATIONS_REGEX.test(file))
```

File: src/commands/dump.js [hunk 2]
```diff
@@ -143,6 +163,25 @@ module.exports = async options => {
...
   }
   const cos = new COS.S3(cosConfig)
+
+  if (ops.annotations){
+    //only download pictures and annotations
+    spinner.message = `Exporting annotations from ${bucket}...`
+    await downloadAnnotations(cos, bucket, 'exported_buckets')
+    
+    if(fs.existsSync(`exported_buckets/${bucket}/_annotations.json`)){
+      fs.renameSync(`exported_buckets/${bucket}/_annotations.json`,`exported_buckets/${bucket}/annotations.json`)
```
Comment (bourdakos1) [inline src/commands/dump.js]: Is there a reason you rename this from `_annotations.json` to `annotations.json`?  This will break `cacli bootstrap`

**Comment by bourdakos1** [comment]
I'm actually not 100% sold on the `cacli export` updates, is it a necessity to make it so granular? I think it adds a lot of complexity and possible confusion? I think just dumping the entire bucket and the option to convert it to the Create ML format (and eventually other formats) should be fine?

File: src/commands/download.js [hunk 3]
```diff
@@ -47,14 +47,20 @@ module.exports = async options => {
   const parser = optionsParse()
   parser.add('model_id')
   parser.add(['--config', '-c'])
+  parser.add([true, '--coreml'])
+  parser.add([true,'--tflite'])
+  parser.add([true,'--web'])
+  parser.add([true,'--graph'])
```
[the remark this answers is not in the document]
Comment (jufabeck2202) [reply src/commands/download.js]: Yes your right, I forgot to remove that

File: src/commands/download.js [hunk 4]
```diff
@@ -47,14 +47,20 @@ module.exports = async options => {
   const parser = optionsParse()
   parser.add('model_id')
   parser.add(['--config', '-c'])
+  parser.add([true, '--coreml'])
+  parser.add([true,'--tflite'])
+  parser.add([true,'--web'])
```
[the remark this answers is not in the document]
Comment (jufabeck2202) [reply src/commands/download.js]: Okay, got it 👍

**Comment by jufabeck2202** [comment]
Yes, you are right about the `cacli export`. I thought it might be useful when you do local training, but you can still export the whole bucket and most of the time you have a separate bucket for pictures and models, so you can still export the annotations. So I removed the export updates 👍

**Comment by bourdakos1** [comment]
Excellent!

**Pull request merged**